### PR TITLE
[Bug Fix] Empty ResouceName results in ill configured UpstreamTLSContext

### PR DESCRIPTION
**Pull request by nschhina**
` rejected: Error adding/updating cluster(s) outbound|443||edition.cnn.com: Proto constraint validation failed (UpstreamTlsContextValidationError.CommonTlsContext: ["embedded message failed validation"] | caused by CommonTlsContextValidationError.CombinedValidationContext: ["embedded message failed validation"] | caused by CombinedCertificateValidationContextValidationError.ValidationContextSdsSecretConfig: ["value is required"]): common_tls_context {
  combined_validation_context {
    default_validation_context {
    }
  }
}`

If caCredential is not specified and no proxy metadata cert override exists, attempt to create empty file watcher result in erroneous TLSContext.  This bug was captured by documentation testing:https://github.com/istio/istio.io/pull/7669. 

This got broken in changes made here: https://github.com/istio/istio/pull/25028

Ref:https://github.com/istio/istio/issues/14039

### CODE REVIEW

**Comment by nschhina** [comment]
cc: @ramaraochavali  @myidpt  @JimmyCYJ

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -1034,12 +1034,24 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
...
 			Sni:              tls.Sni,
 		}
-		tlsContext.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_CombinedValidationContext{
-			CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
-				DefaultValidationContext:         &auth.CertificateValidationContext{MatchSubjectAltNames: util.StringToExactMatch(tls.SubjectAltNames)},
-				ValidationContextSdsSecretConfig: authn_model.ConstructSdsSecretConfig(res.GetRootResourceName(), opts.push.Mesh.SdsUdsPath),
-			},
+
+		// If tls.CaCertificate or CaCertificate in Metadata isn't configured don't set up SdsSecretConfig
+		if res.GetRootResourceName() == "" {
```
Comment (JimmyCYJ) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:1039]: We might need this check under the condition ClientTLSSettings_MUTUAL

Comment (nschhina) [reply pilot/pkg/networking/core/v1alpha3/cluster.go:1039]: yes we do but my concern is if user doesn't specify CA to verify server certs, does istio not supply any of its own to test authenticity of the cert? If yes this case should never happen? Please advice

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -1034,12 +1034,22 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
...
+			tlsContext.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_ValidationContext{
+				ValidationContext: certValidationContext,
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:1041]: shouldn't this be empty not `certValidationContext`? Two concerns are that this *looks* like we are setting a validation context, but I am pretty sure this is always empty when we get to this code path? Other concern is what happens if we send empty validation context to envoy?

**Pull request merged**